services/cd-service/pkg/repository/repository.go

```diff
@@ -218,12 +217,132 @@ func New(ctx context.Context, cfg Config) (Repository, error) {
 			if _, err := state.GetEnvironmentConfigs(); err != nil {
 				return nil, err
 			}
-
+			go result.Work(ctx)
 			return result, nil
 		}
 	}
 }
 
+func (r *repository) Work(ctx context.Context) {
+	defer func() {
+		close(r.queue.elements)
+		for e := range r.queue.elements {
+			e.result <- ctx.Err()
+		}
+	}()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case e := <-r.queue.elements:
+			r.workOnce(e)
+		}
+	}
+}
+
+func (r *repository) applyElements(elements []element, allowFetchAndReset bool) ([]element, error) {
+	for i := 0; i < len(elements); {
+		e := elements[i]
+		applyErr := r.ApplyTransformers(e.ctx, e.transformers...)
+		if applyErr != nil {
+			if errors.Is(applyErr, invalidJson) && allowFetchAndReset {
+				// Invalid state. fetch and reset and redo
+				err := r.FetchAndReset(e.ctx)
+				if err != nil {
+					return elements, err
+				}
+				return r.applyElements(elements, false)
+			} else {
+				e.result <- applyErr
+				elements = append(elements[:i], elements[i+1:]...)
+			}
+		} else {
+			i++
+		}
+	}
+	return elements, nil
+}
+
+var panicError = errors.New("Panic")
+
+func (r *repository) workOnce(e element) {
```

Comment:
maybe `ProcessQueueElement`

Reply:
Since the function drains the queue each time, renaming it as processqueueonce